keanu-project/src/test/java/io/improbable/keanu/algorithms/variational/optimizer/gradient/testcase/SumGaussianTestCase.java

```diff
@@ -0,0 +1,95 @@
+package io.improbable.keanu.algorithms.variational.optimizer.gradient.testcase;
+
+import io.improbable.keanu.algorithms.Variable;
+import io.improbable.keanu.algorithms.variational.optimizer.FitnessFunction;
+import io.improbable.keanu.algorithms.variational.optimizer.FitnessFunctionGradient;
+import io.improbable.keanu.algorithms.variational.optimizer.OptimizedResult;
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.LogLikelihoodFitnessFunctionGradient;
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.LogProbFitnessFunctionGradient;
+import io.improbable.keanu.algorithms.variational.optimizer.nongradient.LogLikelihoodFitnessFunction;
+import io.improbable.keanu.algorithms.variational.optimizer.nongradient.LogProbFitnessFunction;
+import io.improbable.keanu.algorithms.variational.optimizer.nongradient.testcase.NonGradientOptimizationAlgorithmTestCase;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.network.KeanuProbabilisticModelWithGradient;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class SumGaussianTestCase implements GradientOptimizationAlgorithmTestCase, NonGradientOptimizationAlgorithmTestCase {
+
+    private final DoubleVertex A;
+    private final DoubleVertex B;
+
+    private boolean useMLE;
```

Comment:
nit: a few of these variables can be final

Reply:
👍